kong/plugins/httplog/log.lua

```diff
@@ -0,0 +1,19 @@
+local cjson = require "cjson"
+local http_client = require "kong.tools.http_client"
+
+local APPLICATION_JSON_TYPE = "application/json"
+
+local _M = {}
+
+local function log(premature, conf, message)
+  local response, status, headers = http_client.post(conf.logging_url, cjson.encode(message), { ["content-type"] = APPLICATION_JSON_TYPE })
```

Comment:
me loves custom HTTP methods!

yes, `POST` is the defacto choice here, however, there's no reason why a user can dedicate a method for logging (say `LOG`) or use an officially [registered one](http://www.iana.org/assignments/http-methods/http-methods.xhtml) like `REPORT` that's linguistically more appropriate perhaps.

regardless, method naming is contextual to the receiving end and the entity structure, if you think of a log as a *single entity* then sending *entries* should perhaps be done over a `PATCH` request to fit that semantic...

in summary: its best to allow custom methods, and keep a default of `POST`